inc/composer/class-command.php

```diff
@@ -654,6 +661,179 @@ protected function db( InputInterface $input, OutputInterface $output ) {
+				$version = trim( shell_exec( 'mkcert -version' ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed globally already</error>" );
+					return 1;
+				}
+
+				// Check if mkcert is installed locally already, bail if so.
+				$version = trim( shell_exec( "$mkcert -version" ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed locally already</error>" );
+					return 1;
+				}
+
+				exec( "curl -o $mkcert -L https://github.com/FiloSottile/mkcert/releases/download/$mkcert_version/$binary", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not download mkcert binary, try using sudo or manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert $mkcert_version was downloaded.</info>" );
+
+				chmod( $mkcert, 0755);
+
+				exec( "$mkcert -version", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not launch mkcert binary, try manually installing mkcert.</error>" );
+					return 1;
+				}
+				$output->writeln( "<info>mkcert $mkcert_version was installed.</info>" );
+
+				// Setup and accept the root certificate.
+				exec( "$mkcert -install", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( "<error>Could not setup mkcert properly, try manually installing mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert root CA was installed and accepted successfully.</info>" );
+				return 0;
+				break;
+			case 'generate':
+				$mkcert = $this->get_mkcert_binary();
+				if ( ! $mkcert ) {
+					$output->writeln( "<error>mkcert is not installed, run 'composer server ssl install' or install mkcert manually first.</error>" );
+					return 1;
+				}
+
+				// TODO figure out how to programmatically detect the domains to use
+				$domains = $input->getArgument( 'options' )[1] ?? '*.altis.dev';
+
+				exec( "$mkcert -cert-file vendor/ssl-cert.pem -key-file vendor/ssl-key.pem $domains", $dummy, $result );
+
+				if ( $result ) {
+					$output->writeln( "<error>Could not generate certificates! Try generating them manually using mkcert.</error>" );
+					return 1;
+				}
+
+				$output->writeln( "<info>Generated SSL certificate successfully to vendor/ssl-cert.pem and key to vendor/ssl-key.pem.</info>" );
+				break;
+
+			case 'exec':
+				$mkcert = $this->get_mkcert_binary();
+				if ( ! $mkcert ) {
+					$output->writeln( "<error>mkcert is not installed, run 'composer server ssl install' or install mkcert manually first.</error>" );
+					return 1;
+				}
+
+				$command = $input->getArgument( 'options' )[1] ?? null;
+				exec( "$mkcert $command", $exec_output, $result );
+
+				if ( $result ) {
+					$output->writeln( "<error>$exec_output</error>" );
+					return 1;
+				} else {
+					$output->writeln( $exec_output );
+				}
+
+				break;
+
+			case '':
+				$mkcert = $this->get_mkcert_binary();
```

Comment:
There's a fair bit of duplication, I was thinking the `get_mkcert_binary()` can take the `$output` object as an argument but that's overloading its purpose a bit. Perhaps the error message string can just be stored as a variable and reused.

Reply:
Moved the call to the beginning, excluding `install` subcommand, to avoid the duplication.